### PR TITLE
Change the wording in the Filter, Edit, and Export selects as suggested in issue #1480.

### PR DESCRIPTION
**Pull request by drgrice1**
See the referenced issue #1480.

This now also adds a new option to the "Publish" form that if selected allows the user to publish all sets listed on the page.

### CODE REVIEW

File: lib/WeBWorK/ContentGenerator/Instructor/ProblemSetList.pm [hunk 1]
```diff
@@ -927,13 +927,6 @@ sub publish_handler {
 			? CGI::div({ class => 'alert alert-success p-1 mb-0' },
 				$r->maketext("All sets made visible for all students"))
 			: CGI::div({ class => 'alert alert-success p-1 mb-0' }, $r->maketext("All sets hidden from all students"));
-	} elsif ($scope eq "visible") {
```
Comment (taniwallach) [inline lib/WeBWorK/ContentGenerator/Instructor/ProblemSetList.pm:930]: I recommend leaving this block of code, but fixing the messages. @drgrice1 dropped it here as it cannot be reached at present.

Instead I recommend adding `visible` into the list in line 871 (the `values` line near the start of `publish_form`) which determines what options are provided by the select element for the "publish tab". That will allow using the filter feature to select sets which contain a given sub-string (the `match_ids` filter) and then to publish/unpublish the "listed" (matching) sets. 

The same end result can be attained by clicking the "select all" checkbox after using the `match_ids` filter is used and before changing to the "Publish tab" and then using the "selected" option from the pull-down menu of the "Publish" tab.

File: lib/WeBWorK/ContentGenerator/Instructor/ProblemSetList.pm [hunk 2]
```diff
@@ -927,13 +927,6 @@ sub publish_handler {
...
-		@setIDs = @{ $self->{visibleSetIDs} };
-		$result = $value
-			? CGI::div({ class => 'alert alert-success p-1 mb-0' },
-				$r->maketext("All visible sets made visible for all students"))
```
Comment (taniwallach) [inline lib/WeBWorK/ContentGenerator/Instructor/ProblemSetList.pm:934]: $r->maketext("All listed sets were made visible for all the students"))

File: lib/WeBWorK/ContentGenerator/Instructor/ProblemSetList.pm [hunk 3]
```diff
@@ -927,13 +927,6 @@ sub publish_handler {
...
-			: CGI::div({ class => 'alert alert-success p-1 mb-0' },
-				$r->maketext("All visible hidden from all students"));
```
Comment (taniwallach) [inline lib/WeBWorK/ContentGenerator/Instructor/ProblemSetList.pm:936]: `$r->maketext("All listed sets were hidden from all the students"));`

**Review by taniwallach — CHANGES_REQUESTED** [review]
All the "real" changes are to "phrasing" of menu items (and will influence the strings to be translated). That is certainly a change for the better and addresses https://github.com/openwebwork/webwork2/issues/1480.

I recommend letting `publish_handler` process the "listed sets", which can be useful after a `match_ids` filter.

**Comment by taniwallach** [comment]
Note: When testing this I discovered that there is currently a problem with the "filter" for "hidden" or "visible" sets.
When such a filter is used, the code which generates the hidden `visible_sets` input fields is broken, and creates things like
```
<input type="hidden" name="visible_sets" value="ARRAY(0x561d1425d2b8)">
```
where the values should really be the names of the sets which should remain "listed" in the tables.

I showed this to @drgrice1 and he will work on fixing that issue, probably introduced in https://github.com/openwebwork/webwork2/pull/1585 . Until that is fixed, testing of making changes to "listed sets" after filtering by "hidden" or "visible" sets will fail.

**Comment by drgrice1** [comment]
> Note: When testing this I discovered that there is currently a problem with the "filter" for "hidden" or "visible" sets. When such a filter is used, the code which generates the hidden `visible_sets` input fields is broken, and creates things like
> 
> ```
> <input type="hidden" name="visible_sets" value="ARRAY(0x561d1425d2b8)">
> ```
> 
> where the values should really be the names of the sets which should remain "listed" in the tables.
> 
> I showed this to @drgrice1 and he will work on fixing that issue, probably introduced in #1585 . Until that is fixed, testing of making changes to "listed sets" after filtering by "hidden" or "visible" sets will fail.

This is fixed in #1650.

**Comment by drgrice1** [comment]
Note, that this pull request currently only changes wording and does not change any functionality.  The deleted block of code was deleted because the "visible" value did not exist, making that code unreachable as @taniwallach said.  If I merge @taniwallach's pull request to this, then there will be a functionality change.  That is adding a new option to actually export visible (visible as in visible to students) sets.

**Comment by drgrice1** [comment]
@taniwallach:  I merged your pull request.

**Review by taniwallach — APPROVED** [review]
I think this is ready to merge.
Next person who has a look should probably just merge it unless they see something to fix.

**Pull request merged**